x-pack/plugins/dashboard_enhanced/public/services/drilldowns/actions/flyout_edit_drilldown/flyout_edit_drilldown.tsx

```diff
@@ -71,7 +71,7 @@ export class FlyoutEditDrilldownAction implements Action<EmbeddableApiContext> {
   public async isCompatible({ embeddable }: EmbeddableApiContext) {
     if (!isApiCompatible(embeddable)) return false;
     if (getInheritedViewMode(embeddable) !== 'edit') return false;
-    return embeddable.enhancements.dynamicActions.state.get().events.length > 0;
+    return (embeddable.dynamicActionsState$.getValue()?.dynamicActions?.events ?? []).length > 0;
```

Comment:
Using the state container from `DynamicActionManager` was falling out of sync, so the badge beside the "edit" menu option would often reflect the wrong number (especially after the dashboard was reset) - switching to the `dynamicActionsState$` publishing subject, which should be our source of truth, fixed this.